lib/client/db/mcp/errors.go

```diff
@@ -0,0 +1,47 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package mcp
+
+import (
+	"strings"
+
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/trace"
+)
+
+// FormatErrorMessage formats the database MCP error messages.
+// format.
+func FormatErrorMessage(err error) error {
+	switch {
+	// TODO(gabrielcorado): having the database connection error to be the one
+	// from the middleware will make easier and more assertive to determine if
+	// it is a login/session expired error.
+	case strings.Contains(err.Error(), "connection reset by peer") || client.IsErrorResolvableWithRelogin(err):
```

Comment:
i think there are a lot cases you might get "connection reset by peer" and some of them are not resolvable with relogin. which case are we catching here? is it because db connection is cut by proxy?

Reply:
Yes, when the login expires, the local proxy middleware drops the connection right after it is accepted (causing the `reset by peer` error).

I've added logs on this, but I'll see if we can retrieve the error from the middleware and use it here to determine the best user message.

Reply:
instead of guessing, is it possible to actually check if the connection died because of the session expires upon seeing these error?

Reply:
I've added a simple error retriever so we can have accurate error messages, you can check the full change on this commit: https://github.com/gravitational/teleport/pull/54431/commits/770d79bd00019c3091da32d1a36b722a03a86f9f